internal/consumer/partitionConsumer.go

```diff
@@ -256,6 +256,19 @@ func (p *partitionConsumer) messageLoop(offsetRange *kafka.OffsetRange) {
 	}
 }
 
+// delayMsg try to postpone the consumption of the msg for topic.Delay duration if the current timestamp
+// is too soon compare to the msg.Timestamp when Topic.Delay is not zero.
+func (p *partitionConsumer) delayMsg(timestamp time.Time) {
+	delay := p.topicPartition.Delay
+	// check non-zero msg delay
+	if delay > 0 {
```

Comment:
I think you can simplify the logic here by removing the `delay > 0` check by computing `sleepDuration := timestamp + delay - time.Now()`

Reply:
I would say keep delay > 0 might benefit more. for topic setting with Topic.Delay = 0, it will return fast. Also, it will prevent to put the consumer goroutine to sleep when we have delay = 0, and  timestamp - time.Now > 0. Not sure whether it is possible to have msg.Timestamp > time.Now().